temporal-test-server/src/main/java/io/temporal/internal/testservice/SelfAdvancingTimerImpl.java

```diff
@@ -153,10 +152,10 @@ private void runLocked() {
         if (peekedTask != null) {
           timeToAwait = peekedTask.getExecutionTime() - currentTimeMs;
           if (log.isTraceEnabled()) {
-            log.trace("Waiting for {}", Duration.ofMillis(timeToAwait));
+            log.trace("Waiting for {} with outstanding tasks:", Duration.ofMillis(timeToAwait));
             for (TimerTask task : tasks) {
               log.trace(
-                  "Outstanding task: +{} {} {}",
+                  "    +{} <{}>, cancelled: {}",
```

Comment:
This isn't really all that important, but I tend to prefer combining log lines like this that go together into one big one (possibly still with line breaks within it) so that interleaving doesn't lose context.

Given this is test server though, not all that important.